source/Ship.cpp

```diff
@@ -4357,3 +4440,15 @@ double Ship::CalculateDeterrence() const
 		}
 	return tempDeterrence;
 }
+
+
+
+Ship::Escorts::~Escorts()
+{
+	for(const auto &it : list)
+	{
+		shared_ptr<Ship> escort = it.lock();
+		if(escort)
+			escort->parent.reset();
+	}
```

Comment:
Two questions:

1. This would seem more appropriate to do in the Ship destructor no?
2. This seems unnecessary at first glance, because the AI resets the parent pointer if the parent is destroyed and the parent pointer is a weak ptr, so it will be null automatically when the ship is destroyed.